docs/docs/developers/guides/js_apps/how_to_test.md

```diff
@@ -71,29 +101,66 @@ You can also log directly from Aztec contracts. Read [this guide](../local_env/h
+await expect(call.simulate({ from: ownerAddress })).rejects.toThrow(/Balance too low/);
+```
 
 Under the hood, the `send()` method executes a simulation, so we can just call the usual `send().wait()` to catch the same failure.
 
-#include_code local-tx-fails /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+```typescript
+const call = token.methods.transfer(recipientAddress, 200n);
+await expect(call.simulate({ from: ownerAddress })).rejects.toThrow(/Balance too low/);
+```
 
 #### A transaction is dropped
 
 We can have private transactions that work fine locally, but are dropped by the sequencer when tried to be included due to a double-spend. In this example, we simulate two different transfers that would succeed individually, but not when both are tried to be mined. Here we need to `send()` the transaction and `wait()` for it to be mined.
 
-#include_code tx-dropped /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+```typescript
+// Create two transfers that would succeed individually
+const call1 = token.methods.transfer(recipientAddress, 80n);
+const call2 = token.methods.transfer(recipientAddress, 50n);
+
+// Prove both transactions
+const provenCall1 = await call1.prove({ from: ownerAddress });
+const provenCall2 = await call2.prove({ from: ownerAddress });
+
+// First one succeeds
+await provenCall1.send().wait();
+
+// Second one is dropped due to double-spend
+await expect(provenCall2.send().wait()).rejects.toThrow(/dropped|nullifier/i);
+```
 
 #### A public call fails locally
 
 Public function calls can be caught failing locally similar to how we catch private function calls. For this example, we use a [`TokenContract` (GitHub link)](https://github.com/AztecProtocol/aztec-packages/blob/master/noir-projects/noir-contracts/contracts/app/token_contract/src/main.nr) instead of a private one.
 
-#include_code local-pub-fails /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+```typescript
+const call = token.methods.transfer_in_public(ownerAddress, recipientAddress, 1000n, 0);
+await expect(call.simulate({ from: ownerAddress })).rejects.toThrow(/underflow/);
+```
 
 #### A public call fails on the sequencer
 
 This will submit a failing call to the sequencer, who will include the transaction, but without any side effects from our application logic. Requesting the receipt for the transaction will also show it has a reverted status.
 
-#include_code pub-reverted /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+```typescript
+const call = token.methods.transfer_in_public(ownerAddress, recipientAddress, 1000n, 0);
+const receipt = await call.send({ from: ownerAddress }).wait({ dontThrowOnRevert: true });
+
+// Check the transaction was reverted
+expect(receipt.status).toEqual(TxStatus.APP_LOGIC_REVERTED);
+
+// Verify state wasn't modified
+const ownerPublicBalanceSlot = await cheats.aztec.computeSlotInMap(
```

Comment:
I dont think we described how to set up `cheats` yet. Oh I see it is below, it may be worth bringing that section up higher in this doc given cheat codes are referenced